src/popup/components/Modals/PasswordLogin.vue

```diff
@@ -0,0 +1,169 @@
+          :icon="LockIcon"
+          bg-colored
+          bg-dimmed
+          icon-padded
+        />
+      </div>
+      <div class="info">
+        <h3
+          class="text-heading-4 heading"
+          v-text="$t('pages.secureLogin.enterPassword')"
+        />
+
+        <div
+          class="text-description"
+          v-text="$t('pages.secureLogin.toUnlock')"
+        />
+      </div>
+      <form
+        class="password-login-form"
+        @submit.prevent="login"
+      >
+        <InputPassword
+          v-model="password"
+          data-cy="password"
+          autofocus
+          class="password-input"
+          :placeholder="$t('pages.secureLogin.login.placeholder')"
+          :label="$t('pages.secureLogin.login.label')"
+          :message="isAuthFailed ? $t('pages.secureLogin.login.error') : null"
+          @input="isAuthFailed = false"
+        />
+
+        <LinkButton
+          class="forgot-password"
+          href="#"
+          :text="$t('pages.secureLogin.login.forgot')"
+          underlined
+          @click.capture.stop.prevent="openForgotPasswordModal()"
+        />
+
+        <BtnMain
+          class="login-btn"
+          data-cy="login-btn"
+          variant="primary"
+          type="submit"
+          :disabled="!password.length || isAuthenticating || isAuthFailed"
+          :text="$t('pages.secureLogin.login.unlock')"
+          extend
+        />
+      </form>
+    </div>
+  </Modal>
+</template>
+
+<script lang="ts">
+import {
+  ref,
+  defineComponent,
+  PropType,
+} from 'vue';
+
+import type { ResolveCallback } from '@/types';
+import { MODAL_RESET_WALLET } from '@/constants';
+import { useAuth, useModals, useAccounts } from '@/composables';
+
+import Modal from '@/popup/components/Modal.vue';
+import IconBoxed from '@/popup/components/IconBoxed.vue';
+import InputPassword from '@/popup/components/InputPassword.vue';
+import BtnMain from '@/popup/components/buttons/BtnMain.vue';
+import LinkButton from '@/popup/components/LinkButton.vue';
+
+import FingerprintIcon from '@/icons/fingerprint.svg?vue-component';
+import LockIcon from '@/icons/secure-lock.svg?vue-component';
+
+export default defineComponent({
+  components: {
+    Modal,
+    InputPassword,
+    BtnMain,
+    IconBoxed,
+    LinkButton,
+  },
+  props: {
+    resolve: { type: Function as PropType<ResolveCallback>, required: true },
+  },
+  setup(props) {
+    const isAuthFailed = ref(false);
+    const isAuthenticating = ref(false);
+    const password = ref('');
+
+    const { authenticate, isAuthenticated } = useAuth();
+    const { accountsRaw } = useAccounts();
+    const { openModal } = useModals();
+
+    async function openForgotPasswordModal() {
+      await openModal(MODAL_RESET_WALLET, {
+        isResetPassword: true,
+      });
+      // If there are no accounts wallet was reset
+      if (accountsRaw.value.length === 0) {
+        props.resolve();
+      }
+    }
+
+    async function login() {
```

Comment:
I think this modal should not perform the login but only pass the password to the `auth` composable logic. Same as the `SetPassword` component:
```
props.resolve(password.value);
```

Reply:
I researched this and it is more convenient for us to keep the logic here because in order to validate password entered by user we need to try to authenticate without closing the modal.